output/train12_12_23/config12_12_23.txt
# Configure the parameters of our project

import os
import torch
import platform

# The device to be used for training and evaluation (Windows)
if platform.system() == "Darwin":
    if torch.backends.mps.is_available():
        DEVICE = torch.device("cpu")
    else:
        print ("MPS device for MacOS not found.")
    WORKING_DIRECTORY_PATH = "./SICOM_DeepLearning/acceleration_material/"

elif platform.system() == 'Linux':
    WORKING_DIRECTORY_PATH = "/home/conversb/acceleration_material/"
    DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

else:
    DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
    WORKING_DIRECTORY_PATH = ""

# Define the test split which will separate our dataset into train and test
TEST_SPLIT = 0.15

# Define different paths
# The images dataset
IMAGE_DATASET_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "original_images")
# The RGB color masks of the dataset
MASK_DATASET_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "RGB_color_image_masks")
# Labels
LABEL_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "class_dict_seg.csv")
# The output directory
BASE_OUTPUT = os.path.join(WORKING_DIRECTORY_PATH, "output")
# The testing image path
TEST_PATHS = os.path.sep.join([BASE_OUTPUT, "test_paths.txt"])
# The model training plot path
PLOT_TRAIN_PATH = os.path.sep.join([BASE_OUTPUT, "train_plots"])
# The model test plot path
PLOT_TEST_PATH = os.path.sep.join([BASE_OUTPUT, "test_plots"])
# The output serialized model path to save it
MODEL_PATH = os.path.join(BASE_OUTPUT, "unet_tgs_salt.pth")

# List of image types
IMAGE_TYPES = (".jpg", ".jpeg", ".png", ".bmp", ".tif", ".tiff")

# List of gpu
os.environ['CUDA_VISIBLE_DEVICES'] = "0,1,2"

# Vizualization parameters
# True if you want print graphs during the training
MODE_VISUALIZATION = False
VISUALIZATION_DIM = 6

# Test or Train the model
TYPE_PROCESS = "train"  #vallue: {"train", "test"}

# Activate Parallelism
ACTIVATE_PARALLELISM = True
NBR_WORKERS = 24
NBR_GPU = 3

# Define some model parameters
ENC_CHANNELS= (3, 16, 32, 64, 128, 256)
DEC_CHANNELS = (256, 128, 64, 32, 16)
NBR_CLASSES = 24
INPUT_IMAGE_HEIGHT = 256
INPUT_IMAGE_WIDTH = 256
BATCH_SIZE = 64
NUM_EPOCHS = 10
INIT_LR = 0.001 # Initial learning rate
THRESHOLD_TYPE = "mean"
SELECTED_IMAGE_TEST = 10

# If we will be pinning memory during data loading
PIN_MEMORY = True if( DEVICE == "cuda" or DEVICE == "mps") else False


src/config.py
# Configure the parameters of our project
import os
import torch
import platform

# The device to be used for training and evaluation (Windows, MacOS and Linux)
if platform.system() == "Darwin":
    if torch.backends.mps.is_available():
        DEVICE = torch.device("cpu")
    else:
        print ("MPS device for MacOS not found.")
    WORKING_DIRECTORY_PATH = "./SICOM_DeepLearning/Semantic_Segmentation_U-NET/"

elif platform.system() == 'Linux':
    WORKING_DIRECTORY_PATH = "/home/conversb/Semantic_Segmentation_U-NET/"
    DEVICE = "cuda" if torch.cuda.is_available() else "cpu"

else:
    DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
    WORKING_DIRECTORY_PATH = ""

######################################## General Settings ########################################
# Test or Train the model
TYPE_PROCESS = "test"  #value: {"train", "test"}
ID_SESSION = "train_17_12_23_part-7" # unique ID

# Activate Parallelism
ACTIVATE_PARALLELISM = True # Recommended to activate just for Gricad
NBR_WORKERS = 24 # 24 is the maximum for Gricad
# Number or GPU
NBR_GPU = 4 # Adapt this one for the training mode
# Name of GPU's
os.environ['CUDA_VISIBLE_DEVICES'] = "0,1,2,3"

# Data Augmentation
AUG_DATA = True
GENERATE_AUGMENTED_DATA = False # Active just once to generate the data.
#Sometimes there are bugs, so if this is the case, try again and it will be okay.

######################################## Training Settings #######################################
# Define some model parameters
# Define the test split which will separate our dataset into train and test
TEST_SPLIT = 0.15

# Unet Architecture
ENC_CHANNELS= (3, 16, 32, 64)
DEC_CHANNELS = (64, 32, 16)
# Number of class
NBR_CLASSES = 24
ACTIVATE_LABELED_CLASSES = True
# Size of input images
INPUT_IMAGE_HEIGHT = 128
INPUT_IMAGE_WIDTH = 128
BATCH_SIZE = 8
NUM_EPOCHS = 50
# Learning rate
INIT_LR = 0.01
# Threshold just usefull of one class unlabeled
THRESHOLD_TYPE = "mean"

# Vizualization parameters
# True if you want print graphs during the training
MODE_VISUALIZATION = False
VISUALIZATION_DIM = 1 # Lower or egual to NBR_CLASSES

# Early Stopping
EARLY_STOPPING_ACTIVATE = False
PATIENCE = 5

######################################## Test Settings ###########################################
# Number of image to test

SELECTED_IMAGE_TEST = 25

# Metrics
ALL_CONFUSION_MATRIX = True

##################################################################################################

# Define different paths
# The images dataset
IMAGE_DATASET_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "original_images")
# The RGB color masks of the dataset
MASK_DATASET_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "RGB_color_image_masks")
# Labels
LABEL_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "class_dict_seg.csv")
# The output directory
BASE_OUTPUT = os.path.join(WORKING_DIRECTORY_PATH, "output")
# The testing image path
TEST_PATHS = os.path.sep.join([BASE_OUTPUT, ID_SESSION, "test_paths.txt"])
# The model training plot path
PLOT_TRAIN_PATH = os.path.sep.join([BASE_OUTPUT, ID_SESSION, "train_plots"])
# The model test plot path
PLOT_TEST_PATH = os.path.sep.join([BASE_OUTPUT, ID_SESSION, "test_plots"])
# The output serialized model path to save it
MODEL_PATH = os.path.join(BASE_OUTPUT, ID_SESSION, "unet_tgs_salt.pth")
# metric plot path
PLOT_METRICS = os.path.join(BASE_OUTPUT, ID_SESSION, "metrics_plots")
# Path augmented data
AUGMENTED_DATA_IMAGE_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "augmented_data", "images")
AUGMENTED_DATA_MASK_PATH = os.path.join(WORKING_DIRECTORY_PATH, "dataset", "semantic_drone_dataset", "augmented_data", "masks")

# List of image types
IMAGE_TYPES = (".jpg", ".jpeg", ".png", ".bmp", ".tif", ".tiff")

# Work on gricad just with one GPU for testing
if TYPE_PROCESS == "test":
    NBR_GPU = 1

# If we will be pinning memory during data loading
PIN_MEMORY = True if( DEVICE == "cuda" or DEVICE == "mps") else False
